presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastoreConfig.java

```diff
@@ -137,4 +142,47 @@ public GlueHiveMetastoreConfig setCatalogId(String catalogId)
         this.catalogId = Optional.ofNullable(catalogId);
         return this;
     }
+
+    public boolean isUseInstanceCredentials()
+    {
+        return useInstanceCredentials;
+    }
+
+    @Config("hive.metastore.glue.use-instance-credentials")
+    public GlueHiveMetastoreConfig setUseInstanceCredentials(boolean useInstanceCredentials)
+    {
+        this.useInstanceCredentials = useInstanceCredentials;
+        return this;
+    }
+
+    public Optional<String> getAwsCredentialsProvider()
+    {
+        return awsCredentialsProvider;
+    }
+
+    @Config("hive.metastore.glue.aws-credentials-provider")
+    public GlueHiveMetastoreConfig setAwsCredentialsProvider(String awsCredentialsProvider)
+    {
+        this.awsCredentialsProvider = Optional.ofNullable(awsCredentialsProvider);
+        return this;
+    }
+
+    @AssertFalse(message = "No redundant credential allowed")
+    public boolean isRedundantCredentialExists()
+    {
+        int credentialCount = 0;
+        if (!isNullOrEmpty(awsAccessKey.get()) && !isNullOrEmpty(awsSecretKey.get())) {
```

Comment:
use the same pattern as in `getAwsCredentialsProvider`. It only checks if optonal is present.

Reply:
yeah... initially, I use the same pattern ` present `
Later when run test ` TestGlueHiveMetastoreConfig.java#testExplicitPropertyMapping `
At last line assertFullMap, since now only single credential allowed (AssertFalse), I have to set rest of credential as null, but this may contradict to logic in ` assertFullMap ` which may further check whether all field are validated (if field is null exception thrown)... lol ...
So, to work around, I set those field as empty string instead . @sopel39 any alternative to work around ?

Reply:
Did `TestGlueHiveMetastoreConfig.java#testExplicitPropertyMapping` fail when you use same `present` pattern?

If so, I would just remove `isRedundantCredentialExists` for now since we would like validation to follow actual code in Glue Metastore

Reply:
> Did `TestGlueHiveMetastoreConfig.java#testExplicitPropertyMapping` fail when you use same `present` pattern?
> 
> If so, I would just remove `isRedundantCredentialExists` for now since we would like validation to follow actual code in Glue Metastore

if use ` present ` pattern, the test fail because we have to let no more than one credentials present. fullMaping check will fail the comparison, since some value is not set (example: if we set accessKey and secret, we are not allowed to set provider at the same time otherwise, when ` GlueHiveMetastoreConfig ` build through factory, assert logic will fail)
Sure, I am going to remove ` isRedundantCredentialExists ` @sopel39